firmware/ci.c

```diff
@@ -404,23 +419,161 @@ static void video_mode_list(void)
 
 static void video_mode_set(int mode)
 {
-	char mode_descriptors[PROCESSOR_MODE_COUNT*PROCESSOR_MODE_DESCLEN];
+	char mode_descriptor[PROCESSOR_MODE_DESCLEN];
 	if(mode < PROCESSOR_MODE_COUNT) {
-		processor_list_modes(mode_descriptors);
-		wprintf("Setting video mode to %s\r\n", &mode_descriptors[mode*PROCESSOR_MODE_DESCLEN]);
+		processor_describe_mode(mode_descriptor, mode);
+		wprintf("Setting video mode to %s\r\n", mode_descriptor);
 		config_set(CONFIG_KEY_RESOLUTION, mode);
 		processor_start(mode);
 	}
 }
 
+#define NEXT_TOKEN_OR_RETURN(s, t)					\
```

Comment:
Will there every be a case that you need to _not_ skip multiple white space? Happy to change the function, but don't want to cause any other problems.